src/main/java/nl/knaw/dans/lib/util/VersionProvider.java

```diff
@@ -15,21 +15,34 @@
  */
 package nl.knaw.dans.lib.util;
 
-import org.apache.commons.io.IOUtils;
+import lombok.extern.slf4j.Slf4j;
 
-import java.io.IOException;
-import java.io.InputStream;
-import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Optional;
 
+@Slf4j
 public class VersionProvider {
-    public String getVersion() throws IOException {
-        try (InputStream inputStream = getClass().getResourceAsStream("/version.txt")) {
-            if (inputStream != null) {
-                return IOUtils.toString(inputStream, StandardCharsets.UTF_8);
-            }
-            else {
-                throw new IllegalStateException("Version file not found");
+    public String getVersion() {
+        Optional<StackTraceElement> mainOpt = Arrays.stream(
```

Comment:
Maybe simpler: just get the first stackframe, since all applications are started by a main method.

Reply:
the first an the last elements were `java.base/java.lang/Thread.run` and lots in between.